src/pybind/mgr/cephadm/serve.py

```diff
@@ -1212,7 +1212,10 @@ def _purge_deleted_services(self) -> None:
 
             logger.info(f'Purge service {service_name}')
 
-            self.mgr.cephadm_services[spec.service_type].purge(service_name)
+            try:
+                self.mgr.cephadm_services[spec.service_type].purge(service_name)
+            except Exception as ex:
```

Comment:
Again, I suggest catching narrower exception types. For example this could hide a KeyError.

Reply:
My reasoning for the catch all in this specific location was that the cephadm module will fully crash if we raise anything that isn't an OrchestratorError from this spot. The idea was his immediately raises an OrchestratorError which will cause the serve loop to raise a status event that this exception happened (now that I am looking closer, it looks like I did forget to set an event subject on the OrchestratorError being raised, which would invalidate that, but I'll fix that part regardless of what exception type we decide is okay to catch here). I am noticing after some testing, these events don't cause a health warning. You have to explicitly check
```
ceph: root@vm-00 /]# ceph orch ls --service-name nfs.foo -f yaml
service_type: nfs
service_id: foo
service_name: nfs.foo
placement:
  hosts:
  - vm-00
spec:
  port: 2049
status:
  created: '2024-12-16T21:00:34.791598Z'
  ports:
  - 2049
  running: 0
  size: 1
events:
- 2024-12-16T21:01:37.146283Z service:nfs.foo [INFO] "service was created"
- 2024-12-16T21:01:41.111480Z service:nfs.foo [ERROR] "Command ['rados', '-n', 'mgr.vm-00.urkmjh',
  '-k', '/var/lib/ceph/mgr/ceph-vm-00.urkmjh/keyring', '-p', '.nfs', '--namespace',
  'foo', 'rm', 'grace'] got nonzero rc 1 while trying to remove grace file for nfs
  service nfs.foo"

```
to see it, so the visibility isn't great. If we instead created a new health warning type CEPHADM_PURGE_SERVICE_FAILURE or something so that 1) this failure would be very visibily to users and 2) if this ever happened in a teuthology run the log scraper would see it and mark the tests as failed would you be more okay with catching a more general exception type. If not, I can go through and try to find out what exception types we might expect across all the various purge service functions, but I wanted to float the idea of a health warning first.